src/grpc_server.cc

```diff
@@ -2037,6 +2037,17 @@ CommonHandler::SetUpAllRequests()
           std::vector<const TRITONSERVER_Parameter*> const_params;
           for (const auto& param_proto : request.parameters()) {
             if (param_proto.first == "config") {
+              if ((param_proto.second.string_param().find("config") !=
+                   std::string::npos) ||
+                  (param_proto.second.string_param().find("parameters")) !=
+                      std::string::npos) {
```

Comment:
I don't think this is the proper way to check it, what if the user nests "some_other_field"? This check is going to miss that. The question seems to be why the model manager fails to recognize the config is ill-formed and reject the model loading.

Reply:
Seems like I didn't think it thoroughly enough. I have fixed the error checking for the ill-formed config in the core repo. Reverted the previous changes. Please review.